### PR TITLE
Adds support for optional options object in document save method.

### LINKED ISSUE #341: returnNew support
In the doc, https://docs.arangodb.com/3.0/Manual/DataModeling/Documents/DocumentMethods.html#insert, we can see that document.insert can return the newly created document.

In arangojs, collection.insert does not exist, only collection.save. Is this method can support returnNew ?

Regards,
Sylvain

### PR DESCRIPTION
**Pull request by jdpnielsen**
This resolves #341

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2992,6 +2992,24 @@ Creates a new document with the given *data* and returns an object containing th
...
+
+  If *opts* is set, it must be an object with any of the following properties:
+
+  * **waitForSync**: `boolean` (Default: `false`)
+
+    Wait until document has been synced to disk.
+
+  * **returnNew**:  `boolean` (Default: `false`)
+
+    If set to `false`, return additionally the complete new documents under the attribute `new` in the result.
```
Comment (pluma) [inline README.md]: This reads like it should be "If set to `true`". Is this a typo?

**Comment by pluma** [comment]
Just thinking out loud but wouldn't it make more sense from a user's perspective if arangojs simply returned the `new` document instead of the entire wrapper as `_rev` etc are also present on `new`?

File: src/collection.js [hunk 2]
```diff
@@ -647,12 +647,16 @@ class DocumentCollection extends BaseCollection {
     return promise
   }
 
-  save (data, cb) {
+  save (data, opts, cb) {
```
Comment (pluma) [inline src/collection.js]: What would be cool, I think, is if `opts` would be treated as `opts.returnNew` when set to a boolean. Any thoughts?

**Comment by pluma** [comment]
Thank you for the PR. For legal reasons we need a signed Contributor License Agreement (CLA) before we can merge your code into the project.

You can download the form from our website: https://www.arangodb.com/documents/cla.pdf

Once you've filled out and signed it, please send it to us via e-mail: cla@arangodb.com or fax: +49-221-2722999-88.

File: src/collection.js (hunk 2 above)
Comment (jdpnielsen) [inline src/collection.js]: That would definitely be cool. Ill get a commit going.

File: README.md (hunk 1 above)
Comment (jdpnielsen) [inline README.md]: Definitely a typo.

File: src/collection.js (hunk 2 above)
Comment (pluma) [inline src/collection.js]: Awesome. Thanks!

**Comment by jdpnielsen** [comment]
CLA incoming. I don't know about the 'new' field - i'm just assumed it should follow the syntax of the [http docs](https://docs.arangodb.com/3.1/HTTP/Document/WorkingWithDocuments.html#create-document),

File: README.md (hunk 1 above)
Comment (jdpnielsen) [inline README.md]: I copied the statement from the [collection.bulkupdate docs](https://github.com/arangodb/arangojs#collectionbulkupdate) -  should probably also be fixed there?

**Comment by jdpnielsen** [comment]
Do i need to do something regarding backwards compatibility? Tests seem to fail on arangodb 2.8. Probably due to the change from `POST /_api/document?collection={collection}` to `POST /_api/document/{collection}`.

**Comment by pluma** [comment]
Check `collection.js` for `this._connection.arangoMajor >= 3`, that should give you an idea how to adjust the logic to work with both major versions. I'm not sure how to deal with the options when running in 2.8 mode, though. There are similar checks in some of the tests (usually to make the tests simply not register in 2.8).

**Comment by pluma** [comment]
Thanks a ton for this PR. Good work!

**Pull request merged**